drake/core/Path.h.in

```diff
@@ -2,7 +2,7 @@
 
 #include <string>
 
-namespace Drake {
+namespace {
```

Comment:
If this is just to keep an implementation detail out of the Drake namespace, a common approach is:
```c++
namespace Drake {
    namespace detail {
    }
}
```
Would that work here?